src/proto/single/mod.rs

```diff
@@ -162,6 +165,13 @@ impl JobBuilder {
         self
     }
 
+    /// Sets arbitrary key-value pairs to this job's custom data hash.
+    pub fn add_to_custom_data(&mut self, k: String, v: impl Into<serde_json::Value>) -> &mut Self {
```

Comment:
I'd probably make `k: impl Into<String>` as well to avoid all the `.into()`s in the calls to this function.